phpBB/phpbb/profilefields/manager.php

```diff
@@ -499,4 +515,170 @@ public function build_insert_sql_array($cp_data)
+		$this->config_text->delete($profilefield_type_name . '.saved');
+
+		// Get the list of all profile fields of this type
+		$profile_fields = $this->list_profilefields($profilefield_type_name);
+
+		// If no profile fields exist, then nothing to do
+		if (!sizeof($profile_fields))
+		{
+			return;
+		}
+
+		$this->db->sql_transaction('begin');
+
+		// Delete entries from all profile field definition tables
+		$where = $this->db->sql_in_set('field_id', array_keys($profile_fields));
+		$this->db->sql_query('DELETE FROM ' . PROFILE_FIELDS_TABLE . ' WHERE ' . $where);
+		$this->db->sql_query('DELETE FROM ' . PROFILE_FIELDS_LANG_TABLE . ' WHERE ' . $where);
+		$this->db->sql_query('DELETE FROM ' . PROFILE_LANG_TABLE . ' WHERE ' . $where);
+
+		// Drop columns from the Profile Fields data table
+		foreach ($profile_fields as $field_ident)
+		{
+			$this->db_tools->sql_column_remove(PROFILE_FIELDS_DATA_TABLE, 'pf_' . $field_ident);
+		}
+
+		// Reset the order of the remaining fields
+		$order = 0;
+
+		$sql = 'SELECT *
+			FROM ' . PROFILE_FIELDS_TABLE . '
+			ORDER BY field_order';
+		$result = $this->db->sql_query($sql);
+
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			$order++;
+			if ($row['field_order'] != $order)
+			{
+				$sql = 'UPDATE ' . PROFILE_FIELDS_TABLE . "
+					SET field_order = $order
+					WHERE field_id = {$row['field_id']}";
+				$this->db->sql_query($sql);
+			}
+		}
+		$this->db->sql_freeresult($result);
+
+		$this->db->sql_transaction('commit');
+
+		// Log activity
+		foreach ($profile_fields as $field_ident)
+		{
+			add_log('admin', 'LOG_PROFILE_FIELD_REMOVED', $field_ident);
+		}
+	}
+
+	/**
+	* Enable the profile fields of a certain type
+	*
+	* This should be called when an extension which has profile field types
+	* that was disabled is re-enabled so that all those profile fields that
+	* were disabled are enabled again
+	*
+	* @param string $profilefield_type_name Type identifier of the profile fields
+	*/
+	public function enable_profilefields($profilefield_type_name)
+	{
+		// Read the modified information saved on disable from a config_text field to recover values
+		$profile_fields = $this->config_text->get($profilefield_type_name . '.saved');
+
+		// If nothing saved, then nothing to do
+		if (empty($profile_fields))
+		{
+			return;
+		}
+
+		$profile_fields = json_decode($profile_fields, true);
+
+		// Restore the affected profile fields to "active"
+		$sql = 'UPDATE ' . PROFILE_FIELDS_TABLE . '
+			SET field_active = 1
+			WHERE field_active = 0
+				AND ' . $this->db->sql_in_set('field_id', array_keys($profile_fields));
+		$this->db->sql_query($sql);
+
+		// Remove the information saved in the config_text field, not needed any longer
+		$this->config_text->delete($profilefield_type_name . '.saved');
+
+		// Log activity
+		foreach ($profile_fields as $field_ident)
+		{
+			add_log('admin', 'LOG_PROFILE_FIELD_ACTIVATE', $field_ident);
+		}
```

Comment:
`			$this->log->add('admin', $this->user->data['user_id'], (empty($this->user->ip)) ? '' : $this->user->ip, 'LOG_PROFILE_FIELD_ACTIVATE', time(), array($field_ident));`